### PR TITLE
Job auto tenancy

### PR DESCRIPTION
**Pull request by vpratfr**
Implement automatic website ID serialization as per M. Said article on https://divinglaravel.com/preparing-jobs-for-queue

### CODE REVIEW

**Comment by luceos** [comment]
fixes #576

File: tests/unit-tests/Providers/TenancyProviderTest.php [hunk 1]
```diff
@@ -71,7 +72,7 @@ public function current_website_id_is_included_in_notification_job_payload()
     {
         Event::fake();
 
-        $user = $this->userStory()->create();
```
Comment (luceos) [inline tests/unit-tests/Providers/TenancyProviderTest.php]: I was so confused about this one @vpratfr, I almost thought I missed a new feature in Laravel 😊

**Comment by luceos** [comment]
One little request. Can you move the logic inside the TenancyProvider to a QueueProvider in the tenants subdirectory, so that it matches the project structure?

Have you been able to fix the tests?

**Comment by ArlonAntonius** [comment]
One thing we might want to keep in mind on this as well:
What if someone sets the website_id themselves, we shouldn't overwrite it in that case as it would cause things to run on a different tenant than they expect.

**Comment by vpratfr** [comment]
Won't be able to address changes soon, sorry. I have a very busy week coming plus winter holidays

**Comment by vpratfr** [comment]
:thumb:

**Comment by ArlonAntonius** [comment]
Just realized that this TenancyProvider uses the Website model directly, the one we define in the package.
We should refrain from using this as we provide a way for others to use their own models.

**Comment by vpratfr** [comment]
True. The config model shall be used indeed.

**Pull request merged**